apps/emqx/src/emqx_session_mem.erl

```diff
@@ -637,10 +637,10 @@ do_retry_delivery(
             _ = emqx_session_events:handle_event(ClientInfo, {expired, Msg}),
             {Acc, emqx_inflight:delete(PacketId, Inflight)};
         false ->
-            Msg1 = without_inflight_insert_ts(emqx_message:set_flag(dup, true, Msg)),
+            Msg1 = emqx_message:set_flag(dup, true, Msg),
             Update = Data#inflight_data{message = Msg1, timestamp = Now},
             Inflight1 = emqx_inflight:update(PacketId, Update, Inflight),
```

Comment:
Messages stored in Inflight must keep insert timestamps, otherwise, querying inflight will be broken.